src/VisibilityDetectingView.tsx

```diff
@@ -1,35 +1,33 @@
-// @flow
-
 import { Dimensions, UIManager, View } from 'react-native';
 import React, { PureComponent } from 'react';
 import type { ElementRef } from 'react';
 import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';
 
 const TICK_INTERVAL = 100;
 
-type Props = {|
-  children: any,
-  onInvisible: ?() => void,
-  onVisible: ?() => void,
-  style: ?ViewStyleProp,
-|};
+type Props = {
+  children: any;
+  onInvisible: () => void | null | undefined;
+  onVisible: () => void | null | undefined;
+  style: ViewStyleProp | null | undefined;
+};
 
 /** A view that lets you know when its contents become visible/invisible in the screen.
  *  Useful for progressively loading content in a scroll view.
  *  Uses a timer internally to periodically check whether or not it is visible/invisible
-*/
+ */
 export default class VisibilityDetectingView extends PureComponent<Props> {
   previouslyVisible = false;
 
-  tickInterval: ?IntervalID;
+  tickInterval: number | null | undefined;
 
   unmounted = false;
 
-  view: ?ElementRef<typeof View>;
+  view: ElementRef<typeof View> | null | undefined;
 
-  onRef = (view: ?ElementRef<typeof View>) => {
+  onRef = (view?: ElementRef<typeof View> | null) => {
```

Comment:
There's a practical difference here: you can call `onRef(null)` with this signature, which otherwise won't be possible. 

If there are any such calls made, we'd have to find and change them, again in a subsequent cleanup step.